Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -12,28 +12,337 @@
+}
+
+private func expandPeerMacroCodeItem(
+  definition: PeerMacro.Type,
+  attributeNode: AttributeSyntax,
+  attachedTo: DeclSyntax,
+  in context: some MacroExpansionContext,
+  indentationWidth: Trivia
+) throws -> CodeBlockItemListSyntax? {
+  guard
+    let expanded = expandAttachedMacro(
+      definition: definition,
+      macroRole: .peer,
+      attributeNode: attributeNode.detach(in: context),
+      declarationNode: attachedTo.detach(in: context),
+      parentDeclNode: nil,
+      extendedType: nil,
+      conformanceList: nil,
+      in: context,
+      indentationWidth: indentationWidth
+    )
+  else {
+    return nil
+  }
+
+  // Separate the peers and the declaration by two newlines.
+  let indentedSource = "\n\n" + expanded.indented(by: attachedTo.indentationOfFirstLine)
+  return "\(raw: indentedSource)"
+}
+
+/// Expand an accessor macro of a declaration that does not have existing
+/// accessors.
+///
+/// See comment in `expandAccessors` for an explanation why we need this.
+private func expandAccessorMacroWithoutExistingAccessors(
+  definition: AccessorMacro.Type,
+  attributeNode: AttributeSyntax,
+  attachedTo: DeclSyntax,
+  in context: some MacroExpansionContext,
+  indentationWidth: Trivia
+) throws -> AccessorBlockSyntax? {
+  guard
+    let expanded = expandAttachedMacro(
+      definition: definition,
+      macroRole: .accessor,
+      attributeNode: attributeNode.detach(in: context),
+      declarationNode: attachedTo.detach(in: context),
+      parentDeclNode: nil,
+      extendedType: nil,
+      conformanceList: nil,
+      in: context,
+      indentationWidth: indentationWidth
+    )
+  else {
+    return nil
+  }
+
+  // `expandAttachedMacro` adds the `{` and `}` to wrap the accessor block.
+  // Since the entire block is indented, we start it on a new line.
+  let indentedSource = "\n" + expanded.indented(by: attachedTo.indentationOfFirstLine)
+  return "\(raw: indentedSource)"
+}
+
+/// Expand an accessor macro of a declaration that already has an accessor.
+///
+/// See comment in `expandAccessors` for an explanation why we need this.
+private func expandAccessorMacroWithExistingAccessors(
+  definition: AccessorMacro.Type,
+  attributeNode: AttributeSyntax,
+  attachedTo: DeclSyntax,
+  in context: some MacroExpansionContext,
+  indentationWidth: Trivia
+) throws -> AccessorListSyntax? {
+  guard
+    let expanded = expandAttachedMacro(
+      definition: definition,
+      macroRole: .accessor,
+      attributeNode: attributeNode.detach(in: context),
+      declarationNode: attachedTo.detach(in: context),
+      parentDeclNode: nil,
+      extendedType: nil,
+      conformanceList: nil,
+      in: context,
+      indentationWidth: indentationWidth
+    )
+  else {
+    return nil
+  }
+
+  // Separate the accessor from any existing accessors by two spaces
+  let indentedSource = "\n" + expanded.indented(by: attachedTo.indentationOfFirstLine + indentationWidth)
+  return "\(raw: indentedSource)"
```

Comment:
I think the general style is 

```swift
var x: Int {
  get {
    return 2
  }
  set {
    print("new value")
  }
}
```

rather than the following.

```swift
var x: Int {
  get {
    return 2
  }

  set {
    print("new value")
  }
}
```

So yes, I think no extra line should be here.